daliuge-translator/dlg/dropmake/pgt.py

```diff
@@ -262,9 +262,8 @@ def to_pg_spec(
             nm_list = node_list
         nm_len = len(nm_list)
         logger.info(
-            "Drops count: {0}, partitions count: {1}, nodes count: {2}, island count: {3}".format(
-                len(drop_list), num_parts, nodes_len, len(is_list)
-            )
+            "Drops count: %d, partitions count: %d, nodes count: %d, island count: %d",
```

Comment:
Ahh, which one is the string there? Based on the variable names and surrounding text I thought they were all numbers. I'll have a closer look.

Reply:
To quick to answer... I think I misunderstood your comment at first.

I hadn't really thought about it before. I think I'd rather put `%d` in there when one expects an integer rather than `%s` as a fail-safe mechanism. One could go one step ahead and say "let's just use `%s` for everything" but that somehow doesn't feel right, I guess it breaks the "spirit" of the formatting specifiers. It's true that at runtime the system knows the types of all values, so it should know what to deal with them correctly anyway -- unless there are some corner-cases where this isn't obvious. Also using `d` and `f` allow for finer customisation (#digits, length, etc.) which wouldn't be possible with `s`.